### PR TITLE
ci: fail linting for forEach loops

### LINKED ISSUE #190: Apply more strict eslint rules
#### Reason/Context
We should apply more strict eslint rules to keep it consistant. 

- [x] [New line ending](https://eslint.org/docs/rules/eol-last), sprung from https://github.com/asyncapi/generator-model-sdk/pull/184#discussion_r632280767
- [x] [Curly](https://eslint.org/docs/rules/curly), sprung from https://github.com/asyncapi/generator-model-sdk/pull/184#discussion_r632271883
- [ ] Iteration of array - https://github.com/asyncapi/generator-model-sdk/issues/192

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds an eslint error when forEach is used instead of `for .. of`.

Reasons for this change: https://github.com/github/eslint-plugin-github/blob/main/docs/rules/array-foreach.md

blocked by #218 

**Related issue(s)**
fixes #190

### CODE REVIEW

**Comment by jonaslagoni** [comment]
Waiting for the refactor of the interpreter to finish before this can be merged.

File: src/models/CommonModel.ts [hunk 1]
```diff
@@ -366,14 +366,15 @@ export class CommonModel extends CommonSchema<CommonModel> {
    * @param originalSchema 
    * @param alreadyIteratedModels
    */
+  // eslint-disable-next-line sonarjs/cognitive-complexity
```
Comment (magicmatatjahu) [inline src/models/CommonModel.ts:369]: You probably should use also `// NOSONAR`, because I still see the problem on sonarcloud

Comment (jonaslagoni) [reply src/models/CommonModel.ts:370]: I am not sure it is possible for a function like this 🤔

Comment (magicmatatjahu) [reply src/models/CommonModel.ts:370]: Even if you add this comment after function like `functioName(...) { // NOSONAR` ? :)

Comment (jonaslagoni) [reply src/models/CommonModel.ts:369]: That worked 🍻 Thanks!

**Review by magicmatatjahu — APPROVED** [review]
LGTM!

**Pull request merged**